tests/test_router_bgp.rb

```diff
@@ -229,6 +229,35 @@ def test_get_asnum
     bgp.destroy
   end
 
+  def test_nsr
+    nsr(setup_default)
+    nsr(setup_vrf)
+  end
+
+  def nsr(bgp)
+    if platform == :nexus || !@vrf[/default/]
+      refute(bgp.default_nsr,
```

Comment:
Yes, I think you want to assert_nil for bgp.nsr and bgp.default_nsr outside of the assert_raises block.